botones

Zoldyck98 opened the pull request:
cree el componente CustomButton y todo lo necesario para su correcto funcionamiento, con esto me refiero a su estructura en types y su contenido en i18n/ locales. tambien puse 4 ejemplos de los distintos tipos de botones que podemos usar y agregue una simple funcion de prueba para comprobar su correcto funcionamiento e hice unas pequeñas modificaciones en el css del top bar las cuales ire profundizando hasta realizar los cambios necesarios para que quede como se desea.

Zoldyck98 committed:
create the CustomButton component and everything related to its correct operation

Zoldyck98 committed:
Using the CustomButton component to check its correct operation

Zoldyck98 committed:
style: modify the style of the top bar

Zoldyck98 committed:
Rename 'home' to 'buttons' in the es.json and en.json file to better reflect its contents

vladimir1284 commented on src/views/HomeView.vue:
```diff
@@ -1,118 +1,43 @@
+      :text="t('buttons.buttonIconText.text')" 
+      :type="<ButtonType> t('buttons.buttonIconText.type')"      
+      :Icon="PhoneArrowDownLeftIcon"
+      icon-custom-styles="text-white"
```
Pon por defecto todo en blanco y por lo pornto no necesitamos una prop para cambiar de color

Zoldyck98 replied:
la props la cree para dar difernetes estilos, lo de el color fue una prueba para verificar su funcionamieto, ya lo puse el blanco por defecto, borro las props tambien o las dejos para futuros usos?

vladimir1284 commented on src/views/HomeView.vue:
```diff
@@ -1,118 +1,43 @@
+    <CustomButton 
+      :method="() => console.log('click')"
+      :text="t('buttons.buttonIconText.text')" 
+      :type="<ButtonType> t('buttons.buttonIconText.type')"      
```
Esto de q el tipo esté en las traducciones no me convence. Las props es bueno verlas afuera en la instanciación del objeto

Zoldyck98 replied:
listo, las letras en blanco y el tipo en la instanciacion del objeto

Zoldyck98 committed:
feat: create a new component badge.vue with necessary structure and props
- added props: text, count?, BadgeType and badgeCustomStyles
- added type BadgeType in Types.ts

feat: update button types and default text in CustomButton.vue
- Removed type attribute from translation buttons
- Set button text to default to white
- Updated button instantiation to include text directly in the object

Zoldyck98 committed:
refactor: move component ButtonDark.vue from TopBar to fixed position in bottom right of HomeView

- Removed component from the topbar
- Updated styles to position the component in the bottom right corner of HomeView
- Set the component to be fixed during scrolling for better accessibility

Zoldyck98 committed:
refactor: rename CardTypes.ts to Types.ts

- Changed the filename from CardTypes.ts to Types.ts for better clarity and consistency
- Updated all import statements to reflect the new filename

Zoldyck98 commented:
cree el componente bradge junto con todo lo necesario para su correcto funcionamiento, cambie el CardTypes.ts por Types.ts para que especifique mejor su funcion, saque el buttonDark del TopBar y lo puse en el homeview para que se semeje mas a lo que queremos en el resultado final.

vladimir1284 committed:
Merge remote-tracking branch 'origin/main' into 1-componente-de-tarjeta
